lib/game/handlers/chat.ex

```diff
@@ -135,6 +137,33 @@ defmodule ThistleTea.Game.Chat do
     end)
   end
 
+  def handle_chat(state, a, b, ".speed" <> rest, c) do
+    handle_chat(state, a, b, ".modify speed" <> rest, c)
+  end
+
+  def handle_chat(state, _, _, ".modify" <> params, _) do
+    case params |> String.split(" ", trim: true) do
+      ["speed", rate] ->
+        case Float.parse(rate) do
+          {rate, _} when rate >= 0.1 and rate <= 10.0 ->
+            GenServer.cast(self(), {:modify, :speed, rate})
```

Comment:
Open to ideas on how to organize these modify commands, there's a good number of them that modify player & other entity attributes, and they each have validations that must be applied. We could just have a bunch of handle_chat's, or this split method I'm currently doing, would end up pretty gnarly though! 

Alternatively we can abstract out a chat command parser, where you can configure the command, args, and types and it handles the work for you. Almost like a CLI argument parser.

Reply:
Some ideas:
* could skip validation here and send `{:modify, field, {values}}`
  * for speed, `{:modify, :speed, {rate}}`
  * for faction, `{:modify, :faction, {factionid, flagid, npcflagid, dynamicflagid}}`
  * then validation could be handled in the genserver handle_cast functions
* move all debug commands to separate file, similar to how we did with emotes (chat/debug.ex?, chat/gm_commands.ex?)

Reply:
I like letting the GenServer do the validation, how do you feel about the string / float conversion? In my mind, the GenServer `{:modify, :speed, {rate}}` would take `rate` as a float, but it comes in via chat as a string (of course). Also thoughts on doing this as a `cast` vs `call`? If it's a call, we could handle the response from the GenServer to better attach it to a system message.

Reply:
I feel like it'd be simplest to send everything as a string and let the handler perform the conversions, but not sure.

I'm not sure it can be a call, since all this is actually running in the same process. Right now ThistleTea.Game is started to handle new socket connections, but it's also where all the player state is. I think it'd be good to eventually split, though, to have player be more like mob and game object.

But since they're the same process, might be able to use a function and operate on state rather than passing a message in the first place.